### PR TITLE
Don't crash trying to parse a command line that's a directory

### LINKED ISSUE #12535: Terminal crashes on `wt .`
### Windows Terminal version

1.12.10393.0

### Windows build number

19043.1469

### Other Software

- Command Prompt
- Windows Explorer
screenshot

### Steps to reproduce

1. Open Windows Terminal.
2. Open Command Prompt using a non-Windows Terminal context.
3. Execute `wt .` in Command Prompt.
    (May also happen with Git Bash)

### Expected Behavior

I was expecting to have Windows Terminal open in the current directory.

### Actual Behavior

Windows Terminal crashed and I lost about 30 minutes of work.

### PR DESCRIPTION
**Pull request by ianjoneill**
## Summary of the Pull Request
Prevents a crash that could occur when invoking `wt C:\`

## PR Checklist
* [x] Closes #12535
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
Updates `CascadiaSettings::NormalizeCommandLine()` to check that there are a suitable number of command line arguments to be concatenated together, to prevent accessing an array index in `argv` that doesn't exist.

Also prevents a test flake that could occur in `TerminalSettingsTests::CommandLineToArgvW()`, due to generating an empty command line argument.

## Validation Steps Performed
Added a test, and checked that invoking each of the command lines below behaved as expected:
```
wtd C:\ # Window pops up with [error 2147942405 (0x80070005) when launching `C:\']
wtd C:\Program Files # Window pops up with [error 2147942402 (0x80070002) when launching `C:\Program Files']
wtd cmd # cmd profile pops up
wtd C:\Program Files\Powershell\7\pwsh -WorkingDirectory C:\ # PowerShell profile pops up in C:\
wtd "C:\Program Files\Powershell\7\pwsh" -WorkingDirectory C:\ # PowerShell profile pops up in C:\
wtd . # Window pops up with [error 2147942405 (0x80070005) when launching `.']
```

### CODE REVIEW

**Review by zadjii-msft — APPROVED** [review]
Thanks for hopping on this so fast!

**Comment by zadjii-msft** [comment]
@msftbot make sure @lhecker approves

File: src/cascadia/LocalTests_SettingsModel/TerminalSettingsTests.cpp [hunk 1]
```diff
@@ -85,7 +85,8 @@ namespace SettingsModelLocalTests
         for (int i = 0; i < expectedArgc; ++i)
         {
             const bool useQuotes = static_cast<bool>(rng(2));
-            const auto count = static_cast<size_t>(rng(64));
+            // We need to ensure there is at least one character
+            const auto count = static_cast<size_t>(rng(64) + 1);
```
Comment (lhecker) [inline src/cascadia/LocalTests_SettingsModel/TerminalSettingsTests.cpp:89]: I believe this comment (and change) isn't correct, because `CommandLineToArgvW` should work even if you pass it an empty string.

**Review by lhecker — APPROVED** [review]
Thanks for fixing this!

File: src/cascadia/LocalTests_SettingsModel/TerminalSettingsTests.cpp (hunk 1 above)
Comment (ianjoneill) [reply src/cascadia/LocalTests_SettingsModel/TerminalSettingsTests.cpp:89]: This test errored out for me whenever it generated a command line that had a non-quoted empty argument.

In particular at line 115 - `VERIFY_ARE_EQUAL(expectedArgc, argc);` 

`expectedArgc` would be one more than `argc`.

Comment (ianjoneill) [reply src/cascadia/LocalTests_SettingsModel/TerminalSettingsTests.cpp:89]: e.g. a generated string of `aaaa  bbbb` has an `expectedArgc` of 3, but `CommandLineToArgvW()` counts 2.
